etl/steps/data/garden/eth/2023-03-15/ethnic_power_relations.py

```diff
@@ -0,0 +1,247 @@
+        "egip_groups_count": "sum",
+        "excl_groups_count": "sum",
+        "regaut_groups_count": "sum",
+        "regaut_excl_groups_count": "sum",
+        "regaut_egip_groups_count": "sum",
+        "rlvt_groups_count": "sum",
+        "actv_groups_count": "sum",
+        "lpop_headcount": "sum",
+        "egippop_headcount": "sum",
+        "exclpop_headcount": "sum",
+        "discrimpop_headcount": "sum",
+        "maxexclpop_headcount": "sum",
+        "regautpop_headcount": "sum",
+        "regautexclpop_headcount": "sum",
+        "regautegippop_headcount": "sum",
+        "population": "sum",
+    }
+
+    # Add regional aggregates, by summing up the variables in `aggregations`
+    for region in regions:
+        df = geo.add_region_aggregates(df, region=region, aggregations=aggregations, population=df)
+
+    # Filter dataset by regions to make additional calculations and drop regions in df
+    df_regions = df[df["country"].isin(regions)].reset_index(drop=True)
+    df = df[~df["country"].isin(regions)].reset_index(drop=True)
+
+    # Calculate share variables for regions
+    for var in pop_vars:
+        df_regions[var] = df_regions[f"{var}_headcount"] / df_regions["population"] * 100
+
+    # Calculate variables that have not been aggregated
+    df_regions["legippop"] = df_regions["egippop"] / df_regions["lpop"]
+    df_regions["lexclpop"] = df_regions["exclpop"] / df_regions["lpop"]
+    df_regions["ldiscrimpop"] = df_regions["discrimpop"] / df_regions["lpop"]
+    df_regions["lmaxexclpop"] = df_regions["maxexclpop"] / df_regions["lpop"]
+
+    # Concatenate df with df_regions
+    df = pd.concat([df, df_regions], ignore_index=True)
+
+    # Drop headcount and population columns
+    df = df.drop(columns=["population"] + pop_vars_headcount)
+
+    # Verify index and sort
+    df = df.set_index(["country", "year"], verify_integrity=True).sort_index()
+
+    return df
+
+
+# EPR classifies some countries as both their historical and current name. This is not ideal for some cases where their borders changed completely
+def separate_historical_countries(df: pd.DataFrame) -> pd.DataFrame:
+
+    df["country"] = df["country"].astype(str)
+    # RUSSIA / SOVIET UNION
+    # Change the country name for the Soviet Union to Russia for years less or equal to 1991
+    df.loc[(df["country"] == "Russia (Soviet Union)") & (df["year"] <= 1991), "country"] = "Soviet Union"
+    df.loc[(df["country"] == "Russia (Soviet Union)") & (df["year"] > 1991), "country"] = "Russia"
+
+    # VIETNAM
+    # Drop values for "Vietnam, Democratic Republic of" between years 1954 and 1975
+    df = df[
+        ~((df["country"] == "Vietnam, Democratic Republic of") & (df["year"] >= 1954) & (df["year"] <= 1975))
+    ].reset_index(drop=True)
+
+    # YEMEN
+    # Rename "Yemen (Arab Republic of Yemen)" to "Yemen" for years greater than 1990
+    df.loc[(df["country"] == "Yemen (Arab Republic of Yemen)") & (df["year"] > 1990), "country"] = "Yemen"
+
```

Comment:
I haven't looked into the data, but wouldn't you need to also distinguish West Germany from Germany?

Reply:
Now I did it, thanks. In this case nothing much changes, because there is only one ethnic groups considered in the dataset for any Germany ("Germans")